### DISCUSSION TITLE
Supabase Function with textarray

### DISCUSSION THREAD

**Discussion by brentrobbins** [supabase/supabase#6282] [Questions]
I would like to sync the `raw_app_meta_data->>'providers'` array values in my `public.users` table with the `providers` textarray value.  I have the function and trigger in-place, but all of the following have yet to work:

````
begin
  insert into public.users (id, providers)
  values (new.id, 
  new.raw_app_meta_data->>'providers');
  return new;
end;
````

````
begin
  insert into public.users (id, providers)
  values (new.id, 
  new.raw_app_meta_data->>'providers'::textarray);
  return new;
end;
````

````
begin
  insert into public.users (id, providers)
  values (new.id, 
  new.raw_app_meta_data->>'providers'::text);
  return new;
end;
````

Any idea how to modify this function to work?

**Comment by steve-chavez** [comment]
>  but all of the following have yet to work:

Hm, what were the reported errors?

**Reply by brentrobbins** [reply]
@steve-chavez when I add those functions it prevents new accounts from being created. When I try to create a new (Google Auth) test account it just redirects to the home page with the following added to the url `/?error=server_error&error_description=Database+error+saving+new+user` and a new email + password account return 500 error with the msg `Database error saving new user`

**Reply by brentrobbins** [reply]
So I updated it from a `textarray` to just `text` and the following worked when creating the account:

```
begin
  insert into public.users (id, full_name, avatar_url, email, providers)
  values (new.id, new.raw_user_meta_data->>'full_name', new.raw_user_meta_data->>'avatar_url', new.email, new.raw_app_meta_data->>'providers');
  return new;
end;
```

Any idea how I could write the function to update it? Also isn't this going to be an issue since I changed it back to just `text`?

**Reply by steve-chavez** [reply]
I see, so this is related to auth([GoTrue](https://github.com/supabase/gotrue)). @kangmingtay might be able to help here.
